core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1962,6 +1962,26 @@ private[spark] object Utils extends Logging {
     path
   }
 
+  /**
+   * Updates Spark config with properties from a set of Properties.
+   * Provided properties have the highest priority.
+   */
+  def updateSparkConfigFromProperties(conf: SparkConf, properties: Properties): Unit = {
+    properties.asScala.filter { case (k, v) =>
+      k.startsWith("spark.")
+    }.foreach { case (k, v) =>
+      conf.set(k, v)
+      sys.props.getOrElseUpdate(k, v)
+    }
+  }
+
+  /** Loads properties from a String */
+  def loadPropertiesFromString(input: String) : Properties = {
```

Comment:
Ok I will try to use the same function either the one in SparkSubmitArguments or load. The error message must remain the same one way or another.